utils/containers/hash/hash_array.c

```diff
@@ -0,0 +1,66 @@
+/*
```

Comment:
Can we make this file on the same pattern than the others ? i.e. using bazel template ?

Reply:
It's no need to use the template since the implementation is quite simple, only the `hashxxx_array_new` are separated.